### DISCUSSION TITLE
Are topmost/layered windows allowed?

### DISCUSSION THREAD

**Discussion by sackboy19** [ppy/osu#13475] [Q&A] [answered]
So I recently wanted to develop an external key-overlay/pp counter type thing, but the only way I know how is by using a topmost window using the _WS_EX_LAYERED_ flag and setting its position to _HWND_TOPMOST_ with the windows api. What I wanted to ask was if having that type of window layered on top of the game is allowed or will get detected as some sort of cheat? Sorry if its a dumb question but I just wanna be sure before trying to develop anything because I've seen streamers use pp counters that overlay the game and it seems to be allowed.

**Comment by peppy** [comment, marked as answer]
You are posting this on lazer, where we are generally incorporating said features in the game itself.

If you intended to ask about `osu-stable`, then as long as you are only using it for the specific purpose you mention there should be no issue.

**Reply by sackboy19** [reply]
Oh I'm very sorry about that, I didn't know this was the osu lazer repository! I will definitely make sure to double check next time I post. But anyways thank you so much for being polite and answering!
